### ISSUE TITLE
Inappropriate warnings in the context of write or log methods

### ISSUE THREAD

**Issue by Mary904** [spotbugs/spotbugs#1497]
Spotbugs reported a warning of [UI: Usage of GetResource may be unsafe if class is extended (UI_INHERITANCE_UNSAFE_GETRESOURCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ui-usage-of-getresource-may-be-unsafe-if-class-is-extended-ui-inheritance-unsafe-getresource)  for [the following code](https://github.com/eclipse/jetty.project/blob/1cb0a4978f86f6d0800bb8a4262b7810a253c581/demos/demo-jetty-webapp/src/main/java/com/acme/Dump.java#L780):

```java
pout.write("<td>" + this.getClass().getResource(res) + "</td>");
```

I think it is an Inappropriate warning, since the return value of `this.getClass().getResource(res)` is integrated into a string passed to a `write` or `log` method. The `this.getClass().getResource(res)` won't affect the correctness of the subsequent instructions.
